[MM-28140] e2e/system_console/customization: add T1029

mgdelacroix opened the pull request:
#### Summary
Add T1029: Custom branding text can render markdown text

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28140

saturninoabril commented on e2e/cypress/integration/system_console/site_configuration/customization_spec.js:
```diff
@@ -45,6 +45,94 @@ describe('Customization', () => {
     });
+
+    it('MM-T1029 - Custom branding text can render markdown text', () => {
+        // # Visit emoji system console page
```
Should login again as sysadmin.

saturninoabril commented on e2e/cypress/integration/system_console/site_configuration/customization_spec.js:
```diff
@@ -45,6 +45,94 @@ describe('Customization', () => {
+            cy.get('.backstage-form__footer button').click();
+        });
+
+        // # Make sure necessary field is true
+        cy.apiUpdateConfig({TeamSettings: {EnableCustomBrand: true}});
+        cy.reload();
```
Please move config update after admin login so no need to reload.

saturninoabril commented on e2e/cypress/integration/system_console/site_configuration/customization_spec.js:
```diff
@@ -45,6 +45,94 @@ describe('Customization', () => {
+        cy.url().should('include', '/login');
+
+        cy.get('.signup__markdown div div').should('be.visible').within(() => {
+            // * Ensure emoji has been rendered
+            // ToDo: uncomment after fixing MM-28443
+            //cy.get('span.emoticon').should('have.attr', 'title', ':yay:');
```
Good catch! In the ticket, could you add link to this PR so that anyone who'll on it will be able to enable this step and hopefully help validate if the fix works.  Thanks.

mgdelacroix replied:
Done!

saturninoabril reviewed (changes requested):
Thanks Miguel! Please see comments.

saturninoabril reviewed (approved):
Thanks @mgdelacroix! LGTM, tested and passed.

jespino reviewed (approved):
LGTM

lindy65 commented:
Hi @mgdelacroix - On the [test case](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T1029), I'm see a note under "Objective" that says "Note: Removed verification of custom emoji in custom branding text, as that was not implemented after all."

I tested Custom Branding using a custom emoji and it doesn't render.  Only an existing emoji renders.  There is a ticket open to implement the custom emoji rendering in Custom Branding: https://mattermost.atlassian.net/browse/MM-12657

I'm so sorry the test case was not up-to-date - apologies.

In light of this, is it possible for you to update the code to include using an already existing emoji on the Custom Branding page - e.g. :smile: 

In the meantime, I have updated the test in TM4J to reflect the current status quo

//cc @saturninoabril

mgdelacroix commented:
Thanks @lindy65! Sure thing, I'll update the test to use an already existing emoji :) I noticed the custom emoji failure and [added a comment in the test code](https://github.com/mattermost/mattermost-webapp/pull/6401/files#diff-7a40349e7df8f628a218689980fd6c8bR199) for a check to be added when the functionality is implemented, but I didn't notice that a ticket already existed :man_facepalming:

lindy65 commented:
Thanks @mgdelacroix - sorry for the extra work :(

mgdelacroix commented:
No problem at all! :smile: Added [a check for a default emoji](https://github.com/mattermost/mattermost-webapp/pull/6401/files#diff-7a40349e7df8f628a218689980fd6c8bR204) and updated the comment reference to the open ticket in the `ToDo` comment

mgdelacroix commented:
^ @lindy65

lindy65 reviewed (approved):
Thanks @mgdelacroix - looks good to me 👍
